src/opnsense/mvc/app/models/OPNsense/Auth/User.xml

```diff
@@ -0,0 +1,57 @@
+<model>
+    <mount>/system/user+</mount>
+    <version>1.0.0</version>
+    <items>
+        <user type="ArrayField">
+            <uid type=".\UidField"/>
+            <name type="TextField">
+                <Mask>/^[a-zA-Z0-9\.\-_]{1,32}$/</Mask>
+                <ValidationMessage>A username must contain a maximum of 32 alfanumeric characters</ValidationMessage>
+                <Required>Y</Required>
+                <Constraints>
+                    <check001>
+                        <type>UniqueConstraint</type>
+                        <ValidationMessage>This username already exist.</ValidationMessage>
+                    </check001>
+                </Constraints>
+            </name>
+            <disabled type="BooleanField">
+                <Default>0</Default>
+                <Required>Y</Required>
+            </disabled>
+            <scope type="TextField">
+                <Default>user</Default>
+                <Required>Y</Required>
+            </scope>
+            <expires type=".\ExpiresField"/>
+            <authorizedkeys type=".\StoreB64Field"/>
+            <otp_seed type="TextField"/>
+            <shell type="OptionField">
+                <BlankDesc>Default (none for all but root)</BlankDesc>
+                <OptionValues>
+                    <o0 value="/bin/csh">/bin/csh</o0>
+                    <o1 value="/bin/sh">/bin/sh</o1>
+                    <o2 value="/bin/tcsh">/bin/tcsh</o2>
+                </OptionValues>
+            </shell>
+            <user_dn type="TextField"/>
+            <password type="UpdateOnlyTextField"/>
+            <scrambled_password type="BooleanField"  volatile="true"/>
+            <landing_page type="TextField"/>
+            <comment type="TextField"/>
+            <email type="TextField"/>
```

Comment:
Maybe this could be an email field?

https://docs.opnsense.org/development/frontend/models_fieldtypes.html#emailfield

Reply:
@Monviech good catch :) I'm not sure if the old form validated properly, but I'm fine with adding constraints in this case.

Reply:
Im also curious if most text fields that store additional information could be replaced with a DescriptionField

Reply:
eventually, maybe, but rather skip that in the first release to avoid discussions

Reply:
Okay :)